core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala

```diff
@@ -333,6 +334,16 @@ object JavadocDirective {
   val jdkDependentLinkStyle = if (sys.props.get("java.specification.version").exists(_.startsWith("1."))) LinkStyleFrames else LinkStyleDirect
 
   final val JavadocLinkStyleProperty = raw"""javadoc\.(.*).link_style""".r
+
+  private[markdown] def url(link: String, baseUrl: Url, linkStyle: LinkStyle): Url = {
+    val url = Url(link).base
+    val path = url.getPath.replaceAll("(\\b[a-z]+)\\.", "$1/") + ".html"
```

Comment:
This relies on the naming convention of packages being all-lowercase (which is rarely broken).